### ISSUE TITLE
[BUG] SPIFFS : Upload failed(3): File write failed

### ISSUE THREAD

**Issue by kr15uk** [luc-github/ESP3D#380]
**Describe the bug**
Can't upload index.html.gz
"Update failed(3): File write failed" pop up

**To Reproduce**
After FW flash go to the web interface, choose files, upload.

**Expected behavior**
Expect WEBUI to be uploaded and get button go to WEBUI (like it happens on NodeMCU's of mine)

**Screenshots**

**ESP3D Firmware:**
- ESP core version: 2.5.2
- Version: v2.1.0.b39
- Web UI Version: 2.1
- Wifi mode: STA 
- WebUI version: 2.1
- Any change done in FW NetBIOS disabled

**Target Firmware:**
 - Name: Marlin
 - Version 2

**Board used (please complete the following information):**
 - MCU: ESP8826
 - Name: ESP-01 
 - Flash size: 1M (512K SPIFFS)

**Additional context**
1. Flashed with 512K SPIFFS, if using recommended 128K SPIFFS as per Wiki I get not enough space hence due to the WEBUI 2.1 index.html.gz file now being 127K (shows 110K available).
2. Also, have done esptool erase flash prior flashing FW and [ESP710]FORMAT via serial monitor prior upload.

**Comment by luc-github** [comment]
yes webui is now a bottle neck for 1M Flash. I will work on a splited language version to decrease the size. as now 128k is too tight

**Comment by kr15uk** [comment]
Sorry, looks like Additional content "took over" the ticket issue.

Just to clarify:
1M (128K SPIFFS) - not enough space message.
Not an issue hence there is a workaround (512K SPIFFS + dropped NetBIOS).

The issue is with uploading index.html.gz on 1M (512K SPIFFS) which gives 470KB space.
Gives me generic upload failed error (not "not enough space" error like with 128K SPIFFS).
Please see the screenshot.
Thanks.
![Screenshot 2019-10-03 at 09 28 46](https://user-images.githubusercontent.com/5671041/66120038-df5f4880-e5d1-11e9-8687-623eb4f142da.png)

**Comment by luc-github** [comment]
please try to use 194KB for spiffs.

**Comment by kr15uk** [comment]
![Screenshot 2019-10-03 at 12 13 17](https://user-images.githubusercontent.com/5671041/66122278-34518d80-e5d7-11e9-9113-fce2c47a16a7.png)

And serial output
![Screenshot 2019-10-03 at 12 12 16](https://user-images.githubusercontent.com/5671041/66122304-45020380-e5d7-11e9-8dd8-553adcdfbc47.png)

Steps:
1. sudo esptool.py --chip auto erase_flash
2. Flash with 192K SPIFFS (stock ESP3D, no changes apart uncommented ESP32 line)
3. Boot, connect and [ESP710] for a fresh start
4. Log in hit upload - get an error message
5. [ESP720] -> [ESP710] -> [ESP720]
6. Try another upload - error

Hope this helps.

**Comment by luc-github** [comment]
that is weird. I do not have 1M on hand now  but will try with this size on the one I have. web page may have a problem

**Comment by luc-github** [comment]
hmmm
I have installed the 2.5.2 core - commented the NETBIOS feature
set the following setting in Arduino IDE
![image](https://user-images.githubusercontent.com/8822552/66127537-8538a480-e61e-11e9-8b22-a3215935791c.png)

then compiled and flashed:
`Sketch uses 495882 bytes (59%) of program storage space. Maximum is 827376 bytes.`

then uploaded the index.html.gz

![image](https://user-images.githubusercontent.com/8822552/66127450-46a2ea00-e61e-11e9-91ad-fa318366e512.png)

I cannot reproduce you issue 

I would say it is Puya issue but patch is enabled by default in 2.5.2 - I am dry on this

can you try to upload using https://github.com/esp8266/arduino-esp8266fs-plugin
here the latest: https://github.com/esp8266/arduino-esp8266fs-plugin/releases/download/0.4.0/ESP8266FS-0.4.0.zip

**Comment by kr15uk** [comment]
Yep, ESP8266FS did the trick.
![Screenshot 2019-10-03 at 14 27 09](https://user-images.githubusercontent.com/5671041/66130776-3113cd00-e5ea-11e9-96dc-379a60eed639.png)

Anyway, I've ordered a couple ESP-01's of AliExpress to see if they'll behave the same way, btw I've also tried to flash couple NodeMCU's with the same settings etc and had no problems there, strange.

Could client software be at the blame?!
I'll power up VM with clean OS to see maybe something to do with my system.

Thanks for all your help and I'll keep you posted on different OS and new ESP's.

Kris
